daphne_worker_test/tests/test_runner.rs

```diff
@@ -257,13 +259,31 @@ impl TestRunner {
         }
     }
 
-    pub async fn get_hpke_configs(&self, client: &reqwest::Client) -> [HpkeConfig; 2] {
+    pub async fn get_hpke_configs(
+        &self,
+        version: DapVersion,
+        client: &reqwest::Client,
+    ) -> [HpkeConfig; 2] {
         let raw_leader_hpke_config = self.leader_get_raw_hpke_config(&client).await;
         let raw_helper_hpke_config = self.helper_get_raw_hpke_config(&client).await;
-        [
-            HpkeConfig::get_decoded(&raw_leader_hpke_config).unwrap(),
-            HpkeConfig::get_decoded(&raw_helper_hpke_config).unwrap(),
-        ]
+        match version {
+            DapVersion::Draft02 => [
+                HpkeConfig::get_decoded(&raw_leader_hpke_config).unwrap(),
+                HpkeConfig::get_decoded(&raw_helper_hpke_config).unwrap(),
+            ],
+            _ => [
+                HpkeConfigList::get_decoded(&raw_leader_hpke_config)
+                    .unwrap()
+                    .hpke_configs
+                    .pop()
```

Comment:
The tests currently expect the length of this list to be 1. If in the future we change the implementation of `DahneWorkerConfig`  such that this contract is broken, this might lead to some tricky bugs. What I would do is *enforce the contract*: I.e., if the length list is not 1, then panic.